addons/@fumohouse/world/nodes/character_manager.gd

```diff
@@ -12,22 +12,35 @@ const _FALL_LIMIT := -128.0
 
 var _local_character: Fumo
 
+@onready var _fumo_appearances: FumoAppearances = FumoAppearances.get_singleton()
+
+
+func _ready():
+	_fumo_appearances.active_changed.connect(load_appearance)
+
 
 func _process(delta: float):
 	if (
 		_local_character
 		and CommonUtils.do_game_input(self)
 		and Input.is_action_just_pressed("reset_character")
 	):
-		var appearance := _local_character.appearance_manager.appearance
-		_delete_character(true, func(): _spawn_character(appearance, null))
+		_delete_character(true, _spawn_character)
 
 	if _local_character and _local_character.global_position.y < _FALL_LIMIT:
 		var appearance := _local_character.appearance_manager.appearance
 		_delete_character(true, func(): _spawn_character(appearance, null))
 
 
-func _spawn_character(appearance: Appearance, char_transform: Variant) -> Node3D:
+func load_appearance(appearance: Appearance):
+	if _local_character:
+		_local_character.appearance_manager.appearance = appearance
+		_local_character.appearance_manager.load_appearance()
+
+
+func _spawn_character(
+	appearance: Appearance = _fumo_appearances.active, char_transform: Variant = null
```

Comment:
Please leave the default appearance as `null` and check `if not appearance` in the function body, this syntax is strange. It is not clear whether `active` is evaluated on call or on definition